src/graph/commands/epicurve/epicurve.pas

```diff
@@ -0,0 +1,216 @@
+  VariableLabelOutput: TEpiGetVariableLabelType;
+  ReverseStrata:       Boolean;
+  ByVarName:           UTF8String;
+  box1, box:           Integer;
+  i, colour:           Integer;
+  sTitle:              UTF8String;
+
+begin
+  VariableLabelOutput := VariableLabelTypeFromOptionList(Command.Options, FExecutor.SetOptions);
+  ValueLabelOutput    := ValueLabelTypeFromOptionList(Command.Options, FExecutor.SetOptions);
+  sColor              := ChartColorsFromOptions(Command.Options, FExecutor.SetOptions);
+  VarNames            := Command.VariableList.GetIdentsAsList;
+  StratVariable       := TStringList.Create;
+  ReverseStrata       := Command.HasOption('sd', Opt);
+  if (ReverseStrata) then
+    begin
+      Command.Options.Remove(Opt); // don't pass this to TABLES!
+      if (Varnames.Count = 1) then
+        FOutputCreator.DoInfoShort('!sd ignored with a single variable');
+    end;
+
+  // weight variable not allowed for epicurve
+  WeightVarName := '';
+
+  DataFile := FExecutor.PrepareDatafile(VarNames, VarNames);
+  XVar := Datafile.Fields.FieldByName[VarNames[0]];
+  Chart := FChartFactory.NewChart();
+  HistogramData := THistogram.Create(FExecutor, Command);
+  if (Command.HasOption('interval', Opt)) then
+    HistogramData.Interval := Opt.Expr.AsInteger;
+  if (Varnames.Count = 1) then
+    begin
+      dummyVar := DataFile.NewField(ftInteger);
+      dummyVar.Name := '_dummy4epicurve';
+      Varnames.Add('_dummy4epicurve');
+    end;
+  // Note: this does NOT call CalcTables with stratification
+  T := TTables.Create(FExecutor, FOutputCreator);
+  TableData  := T.CalcTables(Datafile, VarNames,
+                StratVariable, WeightVarName, Command.Options, nilTablesRefMap, nilStatistics).UnstratifiedTable;
+  if (ReverseStrata) then
+    TableData.SortByRowLabel(true);
+  HistogramData.Fill(TableData);
+  HistogramData.HistogramToEpicurve;
+  T.Free;
+  ByVarName := Datafile.Fields.FieldByName[VarNames[1]].GetVariableLabel(VariableLabelOutput);
+  // if dummy var was created, remove it now
+  if (Varnames.IndexOf('_dummy4epicurve') > -1) then
+    Varnames.Delete(Varnames.IndexOf('_dummy4epicurve'));
+
+  HistogramSource := THistogramSource.Create(Chart);
+  HistogramSource.Histogram := HistogramData;
+  BarSeries := TBarSeries.Create(Chart);
+  BarSeries.Source := HistogramSource;
+  BarSeries.Stacked := true;
+  BarSeries.BarWidthPercent := 100;
+  BarSeries.ShowInLegend := false;
+  SeriesStyles := TChartStyles.Create(Chart);
+
+  if (Varnames.Count > 1) then
+    begin
+      LegendSource := TListChartSource.Create(Chart);
+      LegendSeries := TBarSeries.Create(Chart);
+      LegendSeries.Source := LegendSource;
+      LegendStyles := TChartStyles.Create(Chart);
+      LegendSeries.ShowInLegend := true;
+      LegendSeries.Styles := LegendStyles;
+      LegendSeries.Legend.Multiplicity:=lmStyle;
+      LegendSeries.Legend.GroupIndex  := 0;
+      Chart.AddSeries(LegendSeries);
+
+      colour := 0;
+      box1 := 0;
```

Comment:
Careful with generic variable names - it's hard to read and understand the code.

Reply:
colourNum instead of colour?